Fix/wallet for new accounts

yenda opened the pull request:
improves  #8759 Сan't send funds from added account (but you have to relogin first after creating the account, I asked Dmitry if he can fix it in the status-go PR here https://github.com/status-im/status-go/pull/1569)
fix #8760 multiple accounts created when tapping save account quickly (can't reproduce in this PR)
fix #8830 No matching clause:offline (accidently)
fix #8802 universal links after login (because why not)

This PR is a bit big because some cleanup was needed so that I could clearly see how to leverage some perf improvements (1 token balance call instead of all tokens * number of accounts) as well as implementation details (new accounts have to be saved via rpc call not just as a setting to make status-go aware of them) and integrate status-go fix for additional account transaction history before relogin. To sum it up

- clean up some mess with networks and current network being in multiple places in app-db
- clean up node status related code
- clean up transaction history fetching (remove all the event firing every block, only triggers when there is a transaction for the account)
- clean up balance fetching (replace multiple calls by a single one)
- use saveAccount rpc call so that new accounts get transaction history immediatly

### Reviewers

This is actually 12 separate commits that can be reviewed individually, each could be done in a separate PR but since they are kind of sequential it would have been a pita to make 12 PRs, 12 reviews and 12QAs

### Testing

#### Areas that maybe impacted

- network related (tested changing network, adding/removing network, browser simpledapp.eth on testnet, wallet transactions)
- wallet transactions (tested transaction from initial wallet account to newly created wallet account right after creating new account to see if wallet history of the new account updates)

##### Functional

- wallet / transactions
- account recovery
- new account
- networks

##### Non-functional

- network consumption: might not consume less data but might be faster for your ens resolution in chat @asemiankevich because it uses a more efficient way to get token balances so less calls are made at startup



### Steps to test

Exploratory testing of the above mentionned areas. Already played a lot with it on both emulator and OnePlus5T real device

If you send eth transaction between you own accounts transfer might not be in the transaction list for one of the accounts and balance won't update until you relogin. This is a known issue that needs to be fixed in status-go

status: ready

Commit:
Remove :node/status leftovers

Commit:
update status-go

Commit:
move networks out of multiaccount

Commit:
remove semaphores

Commit:
move network-id verification

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 4bcb79be | [#2](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-8832/2/) | 2019-08-23 21:15:01 | ~5 min | `ios` | [:page_facing_up: log](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-8832/2/consoleText) |

Commit:
cleanup fetching transactions

Commit:
use wallet_getTokensBalances to reduce number of rpc calls

Commit:
save account with accounts_saveAccounts rpc call

so that transaction history starts being fetch without having to relogin

Commit:
fix ens registration on ropsten

ropsten is 50STT
also fixed cancellation of transaction

Commit:
[fix #8802] universal links after login

fix #8802
No redirect from universal links after login

Commit:
[fix #8759] Сan't send funds from added account

- save added account in keystore properly so that it can be used
- still needs a fix in status-go so that it works without having to relogin

flexsurfer commented on src/status_im/multiaccounts/create/core.cljs:
```diff
@@ -142,15 +142,15 @@
   [multiaccount]
   [(let [{:keys [publicKey address]}
          (get-in multiaccount [:derived constants/path-default-wallet-keyword])]
-     {:public-key publicKey
```
maybe better to change this on go side? or implement normalization function for data from go side?

yenda replied:
Yes homogenize the apis is one of my goal but here I just did what works

flexsurfer commented on src/status_im/multiaccounts/create/core.cljs:
```diff
@@ -160,30 +160,32 @@
-  (let [[wallet-account {:keys [public-key]} :as subaccounts-data] (prepare-subaccounts-data multiaccount)
-        name (gfycat/generate-gfy public-key)
-        photo-path (identicon/identicon public-key)
+  (let [[wallet-account {:keys [publicKey]} :as subaccounts-data] (prepare-subaccounts-data multiaccount)
```
why subaccounts? not just account?

yenda replied:
Good point I'll change it tomorrow

flexsurfer commented on src/status_im/multiaccounts/login/core.cljs:
```diff
@@ -94,17 +92,58 @@
     {:db (assoc db :browser/browsers browsers)}))
 
+(fx/defn initialize-web3-client-version
+  {:events [::initialize-web3-client-version]}
```
why do we need this?

yenda replied:
I don't know just for the about screen I guess? It is something that was there before.

flexsurfer commented on src/status_im/wallet/core.cljs:
```diff
@@ -141,24 +146,49 @@
 (re-frame/reg-fx
  :wallet/get-tokens-balances
  (fn [{:keys [addresses tokens assets]}]
    ;;TODO not great to have so many calls , should be optimized, there is wallet_getTokensBalances why wouldn't use it?
```
so we can remove this todo now

flexsurfer reviewed (approved):
Eric is a real hacker! :)

asemiankevich commented:
@yenda could you please double check if the fixes are in this branch?  i can repro all the issues listed in PR :(

for instance https://github.com/status-im/status-react/issues/8759 even with relogin
![image](https://user-images.githubusercontent.com/28752642/63690318-fe213100-c814-11e9-93c4-e6dda346b6aa.png)

or universal link, or multitap (on iOS)

yenda commented:
@asemiankevich can you try on Android because I just retried all cases and it worked. Are you trying from fresh install? Is it the latest build?

asemiankevich commented:
@yenda i am using  builds number 13 android 
https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190825-000447-61d799-pr.apk
and number 12 iOS https://i.diawi.com/z6heGa

On android:

1. install status app and create new account
2. profile -> dev mode -> ropsten network
3. dapps -> simpledapp.eth -> faucet ETH
4. wallet -> add new account to multiaccount
5. wallet -> status account -> send some ETH to new account
6. switch to new account -> send some ETH back to Status account

I receive the `Transaction failed` message.

![image](https://user-images.githubusercontent.com/28752642/63760083-59abf700-c8c7-11e9-8054-f2c0827fb29a.png)


Also, on Android -> when adding new account to multiaccount:
1. open wallet -> add account 
2. enter password and multitap `Generate account` several times
3. multitap `Finish` button several times

as a result: several accounts added and error shown

![image](https://user-images.githubusercontent.com/28752642/63760059-5153bc00-c8c7-11e9-9402-7d80f076eebb.png)

churik commented:
@asemiankevich 
Could reproduce issue with multitap on "Generate account" on my Android device.
Also is reproducible if tapping really quicky on IOS (could replicate only from 4 attempt).

But couldn't reproduce issur with `Transaction failed` message - tried on both IOS and Android couple of times

yenda commented:
Ok I'll fix the multitap, for the transaction failed you need to logout and login after creating the new account and before sending the transaction, and also you need to send funds before otherwise balance may not update. These workarounds won't be necessary after status-go is fixed

asemiankevich commented:
> Ok I'll fix the multitap, for the transaction failed you need to logout and login after creating the new account and before sending the transaction, and also you need to send funds before otherwise balance may not update. These workarounds won't be necessary after status-go is fixed

true, then i suggest not to close the issue https://github.com/status-im/status-react/issues/8759  until status go is fixed just in case

churik commented:
@yenda also in this PR in addition to https://github.com/status-im/status-react/issues/8806 this error "null is not an object" is shown when creating new account.
I can add this to https://github.com/status-im/status-react/issues/8806 and fix it in terms of this issue.

annadanchenko commented:
#8830 - can't reproduce anymore using iOS build 12 from this PR, so seems to be fixed (iphone 6s, iOS 12.4)

Commit:
rename subaccounts -> accounts

Commit:
[fix #8760] multiple accounts created

when tapping save account quickly

Signed-off-by: yenda <eric@status.im>

asemiankevich commented:
@yenda seems i can't generate new account after incorrect password input attempt https://ethstatus.testrail.net/index.php?/tests/view/835215

Can you please take a look? I can do this in nightly fine, so steps:

1. install status
2. open wallet
3. tap Add account
4. enter wrong password and generate account -> validation error
5. re enter password to correct one and generate one more time

flexsurfer commented:
https://github.com/status-im/status-go/pull/1569 has been merged
